aws/resource_aws_neptune_cluster_parameter_group.go

```diff
@@ -0,0 +1,263 @@
+	}
+
+	if err := d.Set("parameter", flattenNeptuneParameters(describeParametersResp.Parameters)); err != nil {
+		return fmt.Errorf("error setting neptune parameter: %s", err)
+	}
+
+	resp, err := conn.ListTagsForResource(&neptune.ListTagsForResourceInput{
+		ResourceName: aws.String(arn),
+	})
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", arn)
+	}
+
+	if err := d.Set("tags", tagsToMapNeptune(resp.TagList)); err != nil {
+		return fmt.Errorf("error setting neptune tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsNeptuneClusterParameterGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	d.Partial(true)
+
+	if d.HasChange("parameter") {
+		o, n := d.GetChange("parameter")
+		if o == nil {
+			o = new(schema.Set)
+		}
+		if n == nil {
+			n = new(schema.Set)
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		parameters, err := expandNeptuneParameters(ns.Difference(os).List())
+		if err != nil {
+			return err
+		}
+
+		if len(parameters) > 0 {
+			// We can only modify 20 parameters at a time, so walk them until
+			// we've got them all.
+			for parameters != nil {
+				paramsToModify := make([]*neptune.Parameter, 0)
+				if len(parameters) <= neptuneClusterParameterGroupMaxParamsBulkEdit {
+					paramsToModify, parameters = parameters[:], nil
+				} else {
+					paramsToModify, parameters = parameters[:neptuneClusterParameterGroupMaxParamsBulkEdit], parameters[neptuneClusterParameterGroupMaxParamsBulkEdit:]
+				}
+				parameterGroupName := d.Get("name").(string)
+				modifyOpts := neptune.ModifyDBClusterParameterGroupInput{
+					DBClusterParameterGroupName: aws.String(parameterGroupName),
+					Parameters:                  paramsToModify,
+				}
+
+				log.Printf("[DEBUG] Modify Neptune Cluster Parameter Group: %s", modifyOpts)
+				_, err = conn.ModifyDBClusterParameterGroup(&modifyOpts)
+				if err != nil {
+					return fmt.Errorf("Error modifying Neptune Cluster Parameter Group: %s", err)
+				}
+			}
+			d.SetPartial("parameter")
+		}
+	}
+
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "rds",
+		Region:    meta.(*AWSClient).region,
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("cluster-pg:%s", d.Id()),
+	}.String()
+	if err := setTagsNeptune(conn, d, arn); err != nil {
+		return err
+	} else {
+		d.SetPartial("tags")
+	}
+
+	d.Partial(false)
+
+	return resourceAwsNeptuneClusterParameterGroupRead(d, meta)
+}
+
+func resourceAwsNeptuneClusterParameterGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	input := neptune.DeleteDBClusterParameterGroupInput{
+		DBClusterParameterGroupName: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Deleting Neptune Cluster Parameter Group: %s", d.Id())
+	_, err := conn.DeleteDBClusterParameterGroup(&input)
+	if err != nil {
+		if isAWSErr(err, neptune.ErrCodeDBParameterGroupNotFoundFault, "") {
```

Comment:
The SDK provides a separate constant for cluster parameter groups: `ErrCodeDBClusterParameterGroupNotFoundFault`